tests/integration/component.go

```diff
@@ -118,6 +118,12 @@ func componentTests(args ...string) {
 			Expect(cmpAllList).To(ContainSubstring("cmp-git"))
 			helper.CmdShouldPass("odo", append(args, "delete", "cmp-git", "-f")...)
 		})
+
+		It("should succeed listing catalog components", func() {
+
+			// Since components catalog is constantly changing, we simply check to see if this command passes.. rather than checking the JSON each time.
+			helper.CmdShouldPass("odo", "catalog", "list", "components", "-o", "json")
```

Comment:
Can you match it with some minimum output if the json output is big.

Reply:
Or It can be helpful - https://github.com/Benjamintf1/unmarshalledmatchers. 

its already vendored in our repo, you just need to import it.